crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -407,15 +407,15 @@ where
                     }
                 }
 
-                if let Some(attrs) = attrs {
-                    // if we return early then we wouldn't perform these consistency checks, so we
-                    // need to do them here, and should do them before we process any payload
```

Comment:
this looks very relevant,

unsure we can change the order without violating engine tests

Reply:
we can check for sure, but this is a pure control flow change. if we perform the check before handling the attributes we no longer return early